### PR TITLE
2174297: register: do a simple `strip()` on environment(s) input

### PR DESCRIPTION
**Pull request by ptoscano**
When we prompt the user for multiple environments, in the input string any space is removed before further internal processing.  Most likely this was done to minic the non-multi-environment case that strips the input to ensure it is not empty; sadly this makes more than that, making it impossible to interactively type environment names that contain a space, as it will be removed.

Since the result of `_prompt_for_environment()` is passed to `check_set_environment_names()`, which `strip()`s any environment after splitting the input string by comma, we can simply `strip()` the input string in both multi-environments and non-multi-environments cases: it will be enough to ensure that it is non-empty.

Add a simple test to check a couple of these whitespace situations, i.e. in the middle after a comma and at the end.

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=2174297

### CODE REVIEW

File: src/subscription_manager/cli_command/register.py [hunk 1]
```diff
@@ -439,10 +439,12 @@ def _prompt_for_environment(self):
...
 
         if self.cp.has_capability(MULTI_ENV):
-            environment = input(_("Environments: ")).replace(" ", "")
+            environment = input(_("Environments: "))
         else:
-            environment = input(_("Environment: ")).strip()
+            environment = input(_("Environment: "))
         readline.clear_history()
+        # ensure the input is not empty
+        environment.strip()
```
Comment (DuckBoss) [inline src/subscription_manager/cli_command/register.py:447]: This line needs to be changed to `environment = environment.strip()` to have any effect, no?
Otherwise a user input of only blank spaces will not trigger a re-prompt for the environment.

Comment (ptoscano) [reply src/subscription_manager/cli_command/register.py]: hm right, missed in the last version of the patch - fixed, thanks

**Review by jirihnidek — APPROVED** [review]
I tested it with several cases and it looks good to me. :+1:
